core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/workspace/WorkspaceResourceMonitor.scala

```diff
@@ -0,0 +1,90 @@
+package org.broadinstitute.dsde.rawls.monitor.workspace
+
+import akka.actor.{Actor, Props}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkspaceManagerResourceMonitorRecord.JobType.JobType
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{WorkspaceManagerResourceJobRunner, WorkspaceManagerResourceMonitorRecord}
+import org.broadinstitute.dsde.rawls.dataaccess.{SlickDataSource, WorkspaceManagerResourceMonitorRecordDao}
+import org.broadinstitute.dsde.rawls.monitor.workspace.WorkspaceResourceMonitor._
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
+import scala.util.Failure
+
+
+
+object WorkspaceResourceMonitor extends {
+
+  def props(dataSource: SlickDataSource, jobRunners: List[WorkspaceManagerResourceJobRunner])(implicit
+    executionContext: ExecutionContext
+  ): Props = Props(WorkspaceResourceMonitor(dataSource, jobRunners))
+
+  def apply(dataSource: SlickDataSource, jobRunners: List[WorkspaceManagerResourceJobRunner])(implicit
+    executionContext: ExecutionContext
+  ): WorkspaceResourceMonitor =
+    new WorkspaceResourceMonitor(new WorkspaceManagerResourceMonitorRecordDao(dataSource), jobRunners)
+
+  sealed trait WSMJobMonitorMessage
+
+  case object CheckNow extends WSMJobMonitorMessage
+
+  case class CheckDone(creatingCount: Int) extends WSMJobMonitorMessage
+
+  // case class CheckJob(job: WorkspaceManagerResourceMonitorRecord) extends WSMJobMonitorMessage
+
+}
+
+
+class WorkspaceResourceMonitor(jobDao: WorkspaceManagerResourceMonitorRecordDao,
+                               jobRunners: List[WorkspaceManagerResourceJobRunner]
+)(implicit val executionContext: ExecutionContext)
+    extends Actor
+    with LazyLogging {
+
+
+  val registeredRunners: Map[JobType, List[WorkspaceManagerResourceJobRunner]] = jobRunners.groupBy(_.jobType)
+
+  self ! CheckDone(0) // initial poll 1 minute after init
+
+  override def receive: Receive = {
+    case CheckNow => checkJobs().andThen(res => self ! res.getOrElse(CheckDone(0)))
+    // This monitor is always on and polling, and we want that default poll rate to be low, maybe once per minute.
+    // However, if projects are being created, we want to poll more frequently, say ~once per 5 seconds.
+    case CheckDone(count) if count == 0 =>
+      context.system.scheduler.scheduleOnce(1 minute, self, CheckNow)
+    case CheckDone(_) => context.system.scheduler.scheduleOnce(5 seconds, self, CheckNow)
+
+    // case CheckJob(job) => registeredRunners.getOrElse(job.jobType, List()).foreach(_.run(job))
+    case Failure(t) =>
+      logger.error(s"failure monitoring WSM Job", t)
+      context.system.scheduler.scheduleOnce(1 minute, self, CheckNow)
+    case _ => logger.debug(s"WSMJobMonitor received unknown message")
```

Comment:
Nitpick: This seems like it'd be unexpected or due to a misconfiguration of some kind. Perhaps this logger should warn and log the message type received.